jwst/regtest/test_nircam_image.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+import pytest
+from astropy.io.fits.diff import FITSDiff
+from astropy.table import Table, setdiff
+
+from jwst.pipeline.collect_pipeline_cfgs import collect_pipeline_cfgs
+from jwst.stpipe import Step
+
+
+@pytest.fixture(scope="module")
+def run_pipelines(jail, rtdata_module):
+    """Run stages 1-3 pipelines on NIRCam imaging data."""
+    rtdata = rtdata_module
+    rtdata.get_data("nircam/image/jw42424001001_01101_00001_nrca5_uncal.fits")
+
+    collect_pipeline_cfgs("config")
+
+    # Run detector1 pipeline only on one of the _uncal files
+    args = ["config/calwebb_detector1.cfg", rtdata.input,
+        "--steps.dq_init.save_results=True",
+        "--steps.saturation.save_results=True",
+        "--steps.superbias.save_results=True",
+        "--steps.refpix.save_results=True",
+        "--steps.linearity.save_results=True",
+        "--steps.dark_current.save_results=True",
+        "--steps.jump.save_results=True",
+        "--steps.jump.rejection_threshold=50.0",
+        ]
+    Step.from_cmdline(args)
+
+    # And run image2 pipeline on the produced _rate file
+    rtdata.input = "jw42424001001_01101_00001_nrca5_rate.fits"
+    args = ["config/calwebb_image2.cfg", rtdata.input,
+        "--steps.assign_wcs.save_results=True",
```

Comment:
Could also save the ``flat_field`` step results, since that's before ``photom`` (and hence ahead of the _cal file). And actually, since all ``assign_wcs`` does is attach the gwcs object, without modifying pixel values in any way, testing the ``flat_field`` output would essentially check both ``assign_wcs`` and ``flat_field`` (2 for 1).